server/server.go

```diff
@@ -146,6 +146,7 @@ func NewServer(ctx *Context, stopper *stop.Stopper) (*Server, error) {
 	retryOpts.Closer = stopper.ShouldDrain()
```

Comment:
So if you notice the comment above this, I already added some support for this when creating the server.  However, it appears that the "Closer" option on retryOpts is being overridden explicitly in at least one place by the DistSender, and therefore this custom retryOpts seems to be unnecessary.